libretroshare/src/retroshare/rstokenservice.h

```diff
@@ -204,14 +193,27 @@ class RsTokenService
      * total size of groups
      * @param token
      */
-    virtual void requestServiceStatistic(uint32_t& token) = 0;
+    virtual void requestServiceStatistic(uint32_t& token, const RsTokReqOptions &opts) = 0;
 
 	/*!
 	 * To request statistic on a group
 	 * @param token set to value to be redeemed to get statistic
 	 * @param grpId the id of the group
 	 */
-    virtual void requestGroupStatistic(uint32_t& token, const RsGxsGroupId& grpId) = 0;
+    virtual void requestGroupStatistic(uint32_t& token, const RsGxsGroupId& grpId, const RsTokReqOptions &opts) = 0;
+
+
+    /* Poll */
+
+    /*!
+     * Request the status of ongoing request.
+     * Please use this for polling as much cheaper
+     * than polling the specific service as they might
+     * not return intermediate status information
+     * @param token value of token to check status for
+     * @return the current status of request
+     */
+	virtual GxsRequestStatus requestStatus(const uint32_t token) = 0;

```

Comment:
token is passed by value no need of const there ;)

Reply:
fixed